GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/DiskLayoutUpgradeTests.cs

```diff
@@ -0,0 +1,133 @@
+{
+    public abstract class DiskLayoutUpgradeTests : TestsWithEnlistmentPerTestCase
+    {
+        protected const string PlaceholderListDatabaseDelimiter = "\0";
+
+        protected static readonly string PlaceholderListDatabaseContent = $@"A .gitignore{PlaceholderListDatabaseDelimiter}E9630E4CF715315FC90D4AEC98E16A7398F8BF64
+A Readme.md{PlaceholderListDatabaseDelimiter}583F1A56DB7CC884D54534C5D9C56B93A1E00A2B
+A Scripts{PlaceholderListDatabaseDelimiter}{TestConstants.PartialFolderPlaceholderDatabaseValue}
+A Scripts\RunUnitTests.bat{PlaceholderListDatabaseDelimiter}0112E0DD6FC64BF57C4735F4D7D6E018C0F34B6D
+A GVFS{PlaceholderListDatabaseDelimiter}{TestConstants.PartialFolderPlaceholderDatabaseValue}
+A GVFS\GVFS.Common{PlaceholderListDatabaseDelimiter}{TestConstants.PartialFolderPlaceholderDatabaseValue}
+A GVFS\GVFS.Common\Git{PlaceholderListDatabaseDelimiter}{TestConstants.PartialFolderPlaceholderDatabaseValue}
+A GVFS\GVFS.Common\Git\GitRefs.cs{PlaceholderListDatabaseDelimiter}37595A9C6C7E00A8AFDE306765896770F2508927
+A GVFS\GVFS.Tests{PlaceholderListDatabaseDelimiter}{TestConstants.PartialFolderPlaceholderDatabaseValue}
+A GVFS\GVFS.Tests\Properties{PlaceholderListDatabaseDelimiter}{TestConstants.PartialFolderPlaceholderDatabaseValue}
+A GVFS\GVFS.Tests\Properties\AssemblyInfo.cs{PlaceholderListDatabaseDelimiter}5911485CFE87E880F64B300BA5A289498622DBC1
+D GVFS\GVFS.Tests\Properties\AssemblyInfo.cs
+";
+
+        protected FileSystemRunner fileSystem = new SystemIORunner();
+
+        public abstract int GetCurrentDiskLayoutMajorVersion();
+        public abstract int GetCurrentDiskLayoutMinorVersion();
+
+        protected void PlaceholderDatabaseShouldIncludeCommonLines(string[] placeholderLines)
+        {
+            placeholderLines.ShouldContain(x => x.Contains(this.FilePlaceholderString("Readme.md")));
+            placeholderLines.ShouldContain(x => x.Contains(this.FilePlaceholderString("Scripts\\RunUnitTests.bat")));
+            placeholderLines.ShouldContain(x => x.Contains(this.FilePlaceholderString("GVFS\\GVFS.Common\\Git\\GitRefs.cs")));
+            placeholderLines.ShouldContain(x => x.Contains(this.FilePlaceholderString(".gitignore")));
+            placeholderLines.ShouldContain(x => x == this.PartialFolderPlaceholderString("Scripts"));
+            placeholderLines.ShouldContain(x => x == this.PartialFolderPlaceholderString("GVFS"));
+            placeholderLines.ShouldContain(x => x == this.PartialFolderPlaceholderString("GVFS\\GVFS.Common"));
+            placeholderLines.ShouldContain(x => x == this.PartialFolderPlaceholderString("GVFS\\GVFS.Common\\Git"));
+            placeholderLines.ShouldContain(x => x == this.PartialFolderPlaceholderString("GVFS\\GVFS.Tests"));
+        }
+
+        protected void WriteOldPlaceholderListDatabase()
+        {
+            this.fileSystem.WriteAllText(Path.Combine(this.Enlistment.DotGVFSRoot, GVFSHelpers.PlaceholderListFile), PlaceholderListDatabaseContent);

```

Comment:
On Mac the placeholder list would have had `/` path separators, but when the `MountUpgradesPlaceholderListDatabaseToSQLite` Mac function test calls this method a file will be written with `\` separators.

I think that we might need to run `GVFSHelpers.PlaceholderListFile` through a `Replace` that swaps the separators.

I don't think the other methods that use paths are called on Mac, but it would be worth confirming.